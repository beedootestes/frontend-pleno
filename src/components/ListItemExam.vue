<template>
  <div class="exam-item">
    {{ title }}

    <ChevronRightIcon :size="45" />
  </div>
</template>

<script setup lang="ts">
import ChevronRightIcon from 'vue-material-design-icons/ChevronRight.vue';

  defineProps({
    title: {
      type: String,
      required: true
    }
  })
</script>

<style lang="scss" scoped>
  .exam-item {
    width: 100%;
    border: 1px solid $color-border;
    box-shadow: $shadow 0 0 22px;
    display: flex;
    align-items: center;
    border-radius: 20px;
    justify-content: space-between;
    box-sizing: border-box;
    padding: 20px 5px 20px 20px;
  }
</style>