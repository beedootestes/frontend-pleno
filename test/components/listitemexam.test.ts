import {describe, expect, test} from "vitest";
import ListItemExam from "../../src/components/ListItemExam.vue";
import {mount, VueWrapper} from "@vue/test-utils";


describe('Test ListItemExam.vue', () => {
  test('mount component', async () => {
    expect(ListItemExam).toBeTruthy()

    const wrapper: VueWrapper<ListItemExam> = mount(ListItemExam, {
      props: { title: 'First Exam' }
    })

    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.text()).toContain('First Exam')
  })
})
