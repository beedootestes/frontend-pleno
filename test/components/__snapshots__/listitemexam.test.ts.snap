// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test ListItemExam.vue > mount component 1`] = `"<div data-v-f78d2ac5=\\"\\" class=\\"exam-item\\">First Exam <span data-v-f78d2ac5=\\"\\" aria-hidden=\\"true\\" class=\\"material-design-icon chevron-right-icon\\" role=\\"img\\"><svg fill=\\"currentColor\\" class=\\"material-design-icon__svg\\" width=\\"45\\" height=\\"45\\" viewBox=\\"0 0 24 24\\"><path d=\\"M8.59,16.58L13.17,12L8.59,7.41L10,6L16,12L10,18L8.59,16.58Z\\"><!--v-if--></path></svg></span></div>"`;
